docs/DeveloperGuide.md

```diff
@@ -1,38 +1,558 @@
+
+
+<div style="background-color: #89CFF0; border-left: 6px solid #ffa502; padding: 10px;">
+<strong> Note: </strong> Importing a Gradle project is slightly different from importing a normal Java project.
+</div>
+<br/>
+
+**Verify the setup:**
+- Run the seedu.address.Main and try a few commands.
+- Run the tests to ensure they all pass.
+- Refer to the **Manual Testing** section for more in-depth instructions on testing
+
+
+
+---
+
+<h2>
+<span style="color:orange;">Design</span>
+</h2>
+
+<h4>
+<span style="color:orange;">Architechture</span>
+</h4>
+
+
+![image](diagrams/Architecture.png)
+
+The **_Architecture Diagram_** given above explains the high-level design of
+the App.
+
+Given below is a quick overview of main components and how they interact with
+each other.
+
+**Main components of the architecture**
+
+**`GrandRhombus`** is in charge of the app launch and shut down.
+
+- At app launch, it initializes the other components in the correct sequence, and connects them up with each other.
+- At shut down, it shuts down the other components and invokes cleanup methods where necessary.
+
+The bulk of the app’s work is done by the following three components:
+
+- **`CommandParser`**: Parses user input and chooses the command
+- **`Command`**: The individual commands which are then executed. 
+- **`Data`**: A collection of classes including `User`, `UserData` which loads and saves user's modules (`Mod`) into **`user.txt`** file
+
+*`Resources`* contains the `moduledata.txt` file which serves as the database of all CEG modules. 
+These modules are converted into `Mod`s to be stored in other components.
+
+**How the architecture components interact with each other**
+
+The _Sequence Diagram_ below shows how the components interact with each other from startup to termination.
+![image](diagrams/SequenceDiagram.png)
+Note: Command.execute() may use User depending on which command it is e.g. `/add` adds a module to the User's semesterModules list
+
+The program runs continuously till the user inputs `/exit`, which causes `CommandParser` to return **false** to isRunning.
+
+The sections below give more details of each component.
+
+---
+
+<h4>
+<span style="color:orange; text-decoration:underline;">CommandParser component</span>
+</h4>
+
+The `CommandParser` component consists of the `CommandParser` class. 
+
+The `CommandParser` component,
+
+- processes user input and chooses corresponding command.
+- creates new command parses arguments into the command.
+- executes user commands using the `Command` component.
+- returns `isRunning` which is true for all user inputs except `/exit`
+
+---
+
+<h4>
+<span style="color:orange; text-decoration:underline;">Command component</span>
+</h4>
+
+Here’s a (partial) class diagram of the `Command` component:
```

Comment:
This diagram is illegible. You may consider improving the diagram quality. You can break it down to smaller sections to make it readable.